### ISSUE TITLE
Number of modes calculation

### ISSUE THREAD

**Issue by dkirkby** [LSSTDESC/N5K#20]
I have a questions about [these lines](https://github.com/LSSTDESC/N5K/blob/master/n5k/calculator_base.py#L79) in `calculator_base`:
```
    def get_ells(self):
        return np.unique(np.geomspace(2, 2000, 128).astype(int)).astype(float)

    def get_dlog_ell(self):
        return np.log(2000/2)/128
```
Is the second function consistent with the first?  If the first function just returned `np.geomspace(2, 2000, 128)` it would be close (but should be /127 instead of /128?)

However, with the rounding to the nearest integer, values of dlog_ell are non-uniform and quite a bit larger at low ell:
```
ells = np.unique(np.geomspace(2, 2000, 128).astype(int)).astype(float)
dlog_ell = np.gradient(ells)/ells
plt.plot(ells, dlog_ell, 'r.')
plt.axhline(np.log(2000/2)/127)
```
![dlog_ell](https://user-images.githubusercontent.com/185007/105567101-cc9be600-5ce4-11eb-8d83-c5bef870979b.png)

Since `dlog_ell` is used to calculate the number of modes per ell, which feeds directly into the final chisq calculation, are we perhaps downweighting ell < 10 too much?

**Comment by damonge** [comment]
You're absolutely right. I coded that up too quickly. Will fix ASAP

**Comment by damonge** [comment]
@dkirkby the number of modes calculation should be correct now. Thanks for raising this

**Comment by dkirkby** [comment]
After the fix in #21, the relative chisq weights at low ell look good:
![nmodes](https://user-images.githubusercontent.com/185007/105926506-07f52800-5ff7-11eb-9949-8efacd9a22dc.jpg)

However, the NEW/OLD ratio at high ell is around 5 instead of 1.  Just checking that this is intended, since it increases the chisq values quite a bit (from 7.4 to 20.5 for the Limber approx).

**Comment by dkirkby** [comment]
That plot is:
```
T = n5k.N5KCalculatorTester('tests/config_tester.yml')
T.setup()
dlog_ell = np.log(2000/2)/128 # was T.get_dlog_ell()
oldmodes = (T.ls+0.5) * T.fsky * dlog_ell * T.ls
newmodes = T.get_nmodes()
plt.plot(T.ls, newmodes / oldmodes, '.')
plt.axhline(5.2, c='r', ls='--')
```

**Comment by damonge** [comment]
ah, right, but `newmodes` is missing the fsky factor, right?

**Comment by damonge** [comment]
(there is a missing factor of 1/2 in the new calculation which I missed -- gonna fix that now -- but I think the extra factor is due to fsky)

**Comment by dkirkby** [comment]
Yes, you're right, I took the method name too literally :-)  With fsky included, there is just a factor of 2 discrepancy.

**Comment by dkirkby** [comment]
Perhaps include `fsky` in `get_nmodes()` or rename to `get_nmodes_allsky()`?

**Comment by damonge** [comment]
OK, I'll reopen until we fix this (in #23)
